docs/proposals/19-migrating-to-prometheus-operator.md.md

```diff
@@ -0,0 +1,536 @@
+        could be added once in a single namespace. These "common" dashboards can
+        then be discovered by each Grafana and provisioned.
+
+      - In some cases, dashboards are more "specific" because they are related to
+        a certain Kubernetes version.
+
+      - Contract between dashboards in configmaps and the Grafana sidecar.
+
+        - Label schema: `monitoring.gardener.cloud/dashboard-{seed,shoot,shoot-user}=true`
+
+        - Each common dashboard will be deployed in the `monitoring` namespace
+          as a configmap. If the dashboard should be provisioned by the user
+          Grafana in a shoot cluster it should have the label
+          `monitoring.gardener.cloud/dashboard-shoot-user=true`. For dashboards
+          that should be provisioned in the operator grafana the label
+          `monitoring.gardener.cloud/dashboard-shoot=true` is required.
+
+        - Each specific dashboard will be deployed in the shoot namespace. The
+          configmap will use the same label scheme.
+
+        - The grafana [sidecar][grafana-sidecar] must be [configured][sidecar-configuration] with:
+
+        ```yaml
+          env:
+          - name: METHOD
+            value: WATCH
+          - name: LABEL
+            value: monitoring.gardener.cloud/dashboard-shoot # monitoring.gardener.cloud/dashboard-shoot-user for user grafana
+          - name: FOLDER
+            value: /tmp/dashboards
+          - name: NAMESPACE
+            value: monitoring,<shoot namespace>
+        ```
+
+    - Grafana in the seed
+
+      - There is also a grafana deployed in the seed. This grafana will be
+        configured in a very similar way, except it will discover dashboards
+        with a different label.
+
+      - The seed grafana can discover configmaps labeled with
+        `monitoring.gardener.cloud/dashboard-seed`.
+
+      - The sidecar will be configured in a similar way:
+
+      ```yaml
+        env:
+        - name: METHOD
+          value: WATCH
+        - name: LABEL
+          value: monitoring.gardener.cloud/dashboard-seed
+        - name: FOLDER
+          value: /tmp/dashboards
+        - name: NAMESPACE
+          value: monitoring,garden
+      ```
+
+    - Dashboards can have multiple labels and be provisioned in a seed and/or shoot grafana.
+
+6. Migrating to the new monitoring stack:
+    1. Deploy the [prometheus-operator] and its custom resources.
+    1. Delete the old monitoring-stack.
+    1. Configure `Prometheus` to "reuse" the `pv` from the old prometheus's
+       `pvc`. An init container will be temporarily needed for this migration.
+       This ensures that no data is lost and provides a clean migration.
+    1. Any extensions or monitoring configuration that is not migrated to the [promethues-operator] right away will be collected and added to an `additionalScrapeConfig`. Once all extensions and components have migrated, this can be dropped.
```

Comment:
```suggestion
    1. Any extension or monitoring configuration that is not migrated to the [prometheus-operator] right away will be collected and added to an `additionalScrapeConfig`. Once all extensions and components have migrated, this can be dropped.
```